src/Agent.Listener/SelfUpdater.cs

```diff
@@ -128,7 +128,8 @@ private async Task<bool> UpdateNeeded(string targetVersion, CancellationToken to
             Trace.Info($"Current running agent version is {BuildConstants.AgentPackage.Version}");
             PackageVersion agentVersion = new PackageVersion(BuildConstants.AgentPackage.Version);
 
-            return serverVersion.CompareTo(agentVersion) > 0;
+            // Always return true for newer agent versions to enable auto rollback (this feature was introduced after 2.165.0)
+            return serverVersion.CompareTo(agentVersion) > 0 || agentVersion.CompareTo(new PackageVersion("2.165.0")) > 0;
```

Comment:
Should we have a PackageVersion constant rather than instancing a new one every time we compare?

Reply:
I think we can actually just get rid of it. This will always be true since any new agent we release will have version > 2.165.0. If this code path gets exercised it will be supported.

Got rid of it, also added an option to turn this behavior off. This is useful in 2 cases:
1) When we are testing our version locally (so 2.999.999 doesn't get repeatedly downgraded)
2) When someone pulls the latest release from GitHub and doesn't want us to downgrade them.